client/boxoptions/options.go

```
@@ -3,198 +3,210 @@ package boxoptions
 import (
 	"net/netip"
 
-	"github.com/getlantern/radiance/constant"
-	"github.com/getlantern/radiance/option"
 	C "github.com/sagernet/sing-box/constant"
 	O "github.com/sagernet/sing-box/option"
 	dns "github.com/sagernet/sing-dns"
 	"github.com/sagernet/sing/common/json/badoption"
+
+	"github.com/getlantern/sing-box-extensions/ruleset"
+
+	mutruleset "github.com/getlantern/radiance/client/service/mutable-ruleset"
+	"github.com/getlantern/radiance/constant"
+	"github.com/getlantern/radiance/option"
 )
 
-var BoxOptions = O.Options{
-	Log: &O.LogOptions{
-		Disabled:     false,
-		Level:        "trace",
-		Output:       "lantern.log",
-		Timestamp:    true,
-		DisableColor: true,
-	},
-	DNS: &O.DNSOptions{
-		Servers: []O.DNSServerOptions{
-			{
-				Tag:     "dns-google-dot",
-				Address: "tls://8.8.4.4",
-			},
-			{
-				Tag:     "dns-cloudflare-dot",
-				Address: "tls://1.1.1.1",
+func boxOptions(dataDir, logOutput, splitTunnelTag, splitTunnelFormat string) O.Options {
+	// TODO: handle error
+	splitTunFile, _ := mutruleset.FilePath(dataDir, splitTunnelTag, splitTunnelFormat)
```

Since we're not returning a error/handling here, we could receive the `splitTunFile` as a parameter

---

We would then need to parse `splitTunFile` to get the tag and format (needed [here](https://github.com/getlantern/radiance/blob/e4fb28aca8354163100c17dd4f70ec86001c98af/client/boxoptions/options.go#L208)), and that could also result in an error if the file extension isn't correct. We might just need to return an error here..

---

This is addressed with the recent commit.